src/internet_identity/src/archive.rs

```diff
@@ -201,52 +191,65 @@ async fn create_canister(arg: CreateCanisterArgument) -> CallResult<(CanisterIdR
 
 async fn install_archive(
     archive_canister: Principal,
-    wasm: VerifiableWasm,
+    wasm: VerifiedWasm,
     install_mode: CanisterInstallMode,
-) -> Result<(), DeployArchiveResult> {
+) -> Result<(), String> {
     let settings = ArchiveInit {
         ii_canister: id(),
         max_entries_per_call: 1000,
     };
-    let encoded_arg = candid::encode_one(settings).map_err(|err| {
-        Failed(format!(
-            "failed to encode archive install argument: {:?}",
-            err
-        ))
-    })?;
+    let encoded_arg = candid::encode_one(settings)
+        .map_err(|err| format!("failed to encode archive install argument: {:?}", err))?;
 
     install_code(InstallCodeArgument {
         mode: install_mode,
         canister_id: archive_canister,
-        wasm_module: wasm.get_verified_wasm()?,
+        wasm_module: wasm.0,
         arg: encoded_arg,
     })
     .await
     .map_err(|(code, message)| {
-        Failed(format!(
+        format!(
             "failed to install archive canister! error code: {:?}, message: {}",
             code, message
-        ))
+        )
     })
 }
 
-async fn archive_status(archive_canister: Principal) -> Result<CanisterStatusResponse, String> {
+fn verify_wasm(wasm: Vec<u8>) -> Result<VerifiedWasm, String> {
+    let expected_hash = match state::expected_archive_hash() {
+        None => return Err("archive deployment disabled".to_string()),
+        Some(hash) => hash,
+    };
+
+    let mut hasher = Sha256::new();
+    hasher.update(&wasm);
+    let wasm_hash: [u8; 32] = hasher.finalize().into();
```

Comment:
very minor nit: they're both technically wasm hashes, so maybe `actual_hash` would better convey the meaning